src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -45,6 +45,12 @@ public class StreamReadConstraints
      */
     public static final int DEFAULT_MAX_STRING_LEN = 20_000_000;
 
+    /**
+     * Default setting for maximum name length: see {@link Builder#maxNameLength(int)}
+     * for details.
+     */
+    public static final int DEFAULT_MAX_NAME_LEN = 50_000;
```

Comment:
This seems like a reasonable value. I am curious as to how timing works on unit tests wrt longer names; quad-based handling is likely rather inefficient as size grows (esp. for escaped names and/or multi-byte UTF-8).

But we definitely need to ask on mailing lists to see if someone out there knows of use cases with unusually long names (I expect someone somewhere does).

Reply:
I created a thread in the Google Group - https://groups.google.com/g/jackson-dev